lib/api/apiUtils/object/createAndStoreObject.js

```diff
@@ -77,17 +78,30 @@ function createAndStoreObject(bucketName, bucketMD, objectKey, objMD, authInfo,
     if (request.query && request.query['x-amz-acl']) {
         headers['x-amz-acl'] = request.query['x-amz-acl'];
     }
-    const metadataStoreParams = {
-        objectKey, authInfo, metaHeaders, size,
-        contentType: request.headers['content-type'],
-        cacheControl: request.headers['cache-control'],
-        contentDisposition: request.headers['content-disposition'],
-        contentEncoding: removeAWSChunked(request.headers['content-encoding']),
-        expires: request.headers.expires,
-        headers,
-        log,
-        isDeleteMarker,
-    };
+    if (isDeleteMarker) {
```

Comment:
True. Technically, I think we might be using an abundance of caution here since the headers I excluded from the construction of `metadataStoreParams` for the delete marker (cache control, expires, etc.) should not be defined in object delete / multiobjectdelete requests anyway. 

If we assume that we can just go with using the original object, but I guess it might be good to explicitly exclude them in the off chance someone includes those headers in those requests.